content/articles/introduction-to-hibernate/index.md

```diff
@@ -0,0 +1,282 @@
+significantly reduced.
+
+### Disadvantages of Hibernate
+Though Hibernate offers numerous advantages, it also has certain disadvantages. These disadvantages
+includes:
+
+#### 1.Slower compared to JDBC
+Because Hibernate creates a large number of SQL queries at runtime depending upon the mapping, 
+it is a little slower compared to JDBC.
+
+#### 2.Multiple insertion are not permitted
+Some JDBC-supported queries are not allowed by Hibernate. For instance, using a single query, 
+it is not possible to put several objects into the same table. To insert each item,
+the programmer must create a new query.
+
+#### 3.Not suitable for a small project
+Because a small project has fewer tables, adopting the whole Hibernate framework is more
+time-consuming than beneficial.
+
+#### 4. Batch processing has a poor track record in Hibernate
+For batch processing, it's best to stick with plain JDBC because Hibernate's performance 
+is not great.
+
+#### 5. A lot of effort to learn API 
+Learning Hibernate takes a lot of time and effort. As a result, learning Hibernate is not a simple task.
+
+#### Hibernate Example
+Here's an example of  implementation of Hibernate.It is for one-to-one biderectional mapping.
+
+```java
+ //This is customer.java class
+import java.io.Serializable;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.OneToOne;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+
+@Entity
+public class customer implements Serializable {
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)//to autogenerate the primary key
+	private Long id;
+
+	@Column(name = "customer_name")
+	private String customer_name;
+
+	@Column(name = "custmer_id")
+	private String custmer_id;
+
+	@Column(name = "customer_email")
+	private customer_email;
+
+	@OneToOne(cascade = CascadeType.ALL)//creating a one to one relationship
+	@JoinColumn(name = "location_id", referencedColumnName = "id")//joining  the two tables
+	private Location location;
+	
+}
+//This is location.java class
+//import  the same packages  as the one imported in customer.java class
+@Entity
+public class Location {
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private Long id;
+
+	@Column(name = "state")
+	private String state;
+
+	@Column(name = "state_number")
+	private String state_number;
+
+	@OneToOne(cascade = CascadeType.ALL, mappedBy = "location")
+	private Customer customer;
+
+}
+//main class
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.hibernate.SessionFactory;
+import java.util.List;
+import org.hibernate.cfg.Configuration;
+import com.netsurfingzone.entity.Location;
+import com.netsurfingzone.entity.Customer;

```

Comment:
Why do you have the above imports in the middle of the code? This is not a good programming practice. Move them to the top

Reply:
@WanjaMIKE,I have moved the imports to the  top